src/arch/amdgpu/vega/insts/vop3p.cc

```diff
@@ -675,9 +678,15 @@ Inst_VOP3P__V_PK_FMA_F32::execute(GPUDynInstPtr gpuDynInst)
             uint32_t s2l = (opsel & 4) ? bits(src2[lane], 63, 32)
                                        : bits(src2[lane], 31, 0);
 
-            float dword1 = std::fma(*reinterpret_cast<float*>(&s0l),
-                                    *reinterpret_cast<float*>(&s1l),
-                                    *reinterpret_cast<float*>(&s2l));
+            float s0lf = *reinterpret_cast<float*>(&s0l);
+            float s1lf = *reinterpret_cast<float*>(&s1l);
+            float s2lf = *reinterpret_cast<float*>(&s2l);
+
+            if (neg & 1) s0lf = -s0lf;
```

Comment:
can you add a comment explaining these neg if's?

Reply:
I could add something somewhere, but the input modifiers like neg/neg_hi are common to all vop3p instructions (also regular vop3) so to reduce clutter in the code maybe it makes more sense at the top of the file?

Reply:
yeah, I'm fine with a separate commit for this then.